### ISSUE TITLE
[bug]: ENUM trunks only work with chan_sip

### ISSUE THREAD

**Issue by JoseGoncalves** [FreePBX/issue-tracker#495]
### FreePBX Version

FreePBX 17

### Issue Description

I had a FreePBX 15 system were I used an ENUM trunk to connect to a remote institute. After migrated that system to FreePBX 17 with Asterisk 20 and `chan_sip` disabled, I was unable to communicate with that trunk!
Looking at Asterisk logs I see this when I try to make a call on that trunk:
```
channel.c: No channel type registered for 'sip'
app_dial.c: Unable to create channel of type 'sip' (cause 66 - Channel not implemented)
```
Only after enabling `chan_sip` I was able to restore the calls with it.
So it seems ENUM trunks need to be updated to support `chan_pjsip`, or else, they are useless with Asterisk 21 or 22, were `chan_sip` is unavailable.

### Operating Environment

Debian 12
Asterisk 20.9.3

```
$ fwconsole ma list
No repos specified, using: [standard] from last GUI settings

+---------------------+------------+---------+---------+-----------+
| Module              | Version    | Status  | License | Signature |
+---------------------+------------+---------+---------+-----------+
| accountcodepreserve | 17.0.0.1   | Enabled | GPLv2   | Sangoma   |
| allowlist           | 17.0.1.1   | Enabled | GPLv3+  | Sangoma   |
| amd                 | 17.0.1     | Enabled | GPLv3+  | Sangoma   |
| announcement        | 17.0.2.1   | Enabled | GPLv3+  | Sangoma   |
| api                 | 17.0.1.6   | Enabled | AGPLv3+ | Sangoma   |
| arimanager          | 17.0.1.1   | Enabled | GPLv3+  | Sangoma   |
| asterisk-cli        | 17.0.2     | Enabled | GPLv3+  | Sangoma   |
| asteriskinfo        | 17.0.2     | Enabled | GPLv3+  | Sangoma   |
| backup              | 17.0.5.60  | Enabled | GPLv3+  | Sangoma   |
| blacklist           | 17.0.1.2   | Enabled | GPLv3+  | Sangoma   |
| builtin             |            | Enabled |         | Unsigned  |
| bulkhandler         | 17.0.5     | Enabled | GPLv3+  | Sangoma   |
| calendar            | 17.0.4.20  | Enabled | GPLv3+  | Sangoma   |
| callback            | 17.0.2.1   | Enabled | GPLv3+  | Sangoma   |
| callforward         | 17.0.1.6   | Enabled | AGPLv3+ | Sangoma   |
| callrecording       | 17.0.3.7   | Enabled | AGPLv3+ | Sangoma   |
| callwaiting         | 17.0.3.4   | Enabled | GPLv3+  | Sangoma   |
| cdr                 | 17.0.4.22  | Enabled | GPLv3+  | Sangoma   |
| cel                 | 17.0.2.9   | Enabled | GPLv3+  | Sangoma   |
| certman             | 17.0.3.13  | Enabled | AGPLv3+ | Sangoma   |
| cidlookup           | 17.0.1.1   | Enabled | GPLv3+  | Sangoma   |
| conferences         | 17.0.3.2   | Enabled | GPLv3+  | Sangoma   |
| configedit          | 17.0.1.4   | Enabled | AGPLv3+ | Sangoma   |
| contactmanager      | 17.0.5.11  | Enabled | GPLv3+  | Sangoma   |
| core                | 17.0.17    | Enabled | GPLv3+  | Sangoma   |
| customappsreg       | 17.0.1     | Enabled | GPLv3+  | Sangoma   |
| customcontexts      | 17.0.1.3   | Enabled | GPLv2+  | Sangoma   |
| dashboard           | 17.0.4.5   | Enabled | AGPLv3+ | Sangoma   |
| daynight            | 17.0.1.1   | Enabled | GPLv3+  | Sangoma   |
| dictate             | 17.0.1.2   | Enabled | GPLv3+  | Sangoma   |
| directory           | 17.0.1.1   | Enabled | GPLv3+  | Sangoma   |
| disa                | 17.0.5     | Enabled | AGPLv3+ | Sangoma   |
| donotdisturb        | 17.0.2.3   | Enabled | GPLv3+  | Sangoma   |
| dynroute            | 17.0.3.2   | Enabled | GPLv3+  | Sangoma   |
| extensionsettings   | 17.0.1     | Enabled | GPLv3+  | Sangoma   |
| fax                 | 17.0.3.4   | Enabled | GPLv3+  | Sangoma   |
| featurecodeadmin    | 17.0.2     | Enabled | GPLv3+  | Sangoma   |
| filestore           | 17.0.2.30  | Enabled | AGPLv3  | Sangoma   |
| findmefollow        | 17.0.4.10  | Enabled | GPLv3+  | Sangoma   |
| framework           | 17.0.19.13 | Enabled | GPLv2+  | Sangoma   |
| hotelwakeup         | 17.0.1.6   | Enabled | GPLv2   | Sangoma   |
| iaxsettings         | 17.0.1     | Enabled | AGPLv3  | Sangoma   |
| infoservices        | 17.0.1.1   | Enabled | GPLv2+  | Sangoma   |
| ivr                 | 17.0.8     | Enabled | GPLv3+  | Sangoma   |
| languages           | 17.0.1     | Enabled | GPLv3+  | Sangoma   |
| logfiles            | 17.0.3.3   | Enabled | GPLv3+  | Sangoma   |
| manager             | 17.0.6     | Enabled | GPLv2+  | Sangoma   |
| miscapps            | 17.0.3     | Enabled | GPLv3+  | Sangoma   |
| miscdests           | 17.0.1.1   | Enabled | GPLv3+  | Sangoma   |
| missedcall          | 17.0.1.2   | Enabled | GPLv3+  | Sangoma   |
| music               | 17.0.5     | Enabled | GPLv3+  | Sangoma   |
| outcnam             | 17.0.1     | Enabled | GPLv3+  | Sangoma   |
| outroutemsg         | 17.0.1     | Enabled | GPLv3+  | Sangoma   |
| paging              | 17.0.3     | Enabled | GPLv3+  | Sangoma   |
| parking             | 17.0.2.5   | Enabled | GPLv3+  | Sangoma   |
| phpinfo             | 17.0.1     | Enabled | GPLv2+  | Sangoma   |
| pinsets             | 17.0.3.2   | Enabled | GPLv3+  | Sangoma   |
| pm2                 | 17.0.3.3   | Enabled | AGPLv3+ | Sangoma   |
| presencestate       | 17.0.2.4   | Enabled | GPLv3+  | Sangoma   |
| printextensions     | 17.0.1.2   | Enabled | GPLv3+  | Sangoma   |
| queueprio           | 17.0.1.4   | Enabled | GPLv3+  | Sangoma   |
| queues              | 17.0.1.12  | Enabled | GPLv2+  | Sangoma   |
| recordings          | 17.0.2.2   | Enabled | GPLv3+  | Sangoma   |
| ringgroups          | 17.0.2.6   | Enabled | GPLv3+  | Sangoma   |
| setcid              | 17.0.1.2   | Enabled | GPLv3+  | Sangoma   |
| sipsettings         | 17.0.6.9   | Enabled | AGPLv3+ | Sangoma   |
| soundlang           | 17.0.4.1   | Enabled | GPLv3+  | Sangoma   |
| superfecta          | 17.0.2.1   | Enabled | GPLv2+  | Sangoma   |
| timeconditions      | 17.0.1.18  | Enabled | GPLv3+  | Sangoma   |
| tts                 | 17.0.1.1   | Enabled | GPLv3+  | Sangoma   |
| ttsengines          | 17.0.1     | Enabled | AGPLv3  | Sangoma   |
| userman             | 17.0.6.28  | Enabled | AGPLv3+ | Sangoma   |
| vmblast             | 17.0.2     | Enabled | GPLv3+  | Sangoma   |
| voicemail           | 17.0.5.22  | Enabled | GPLv3+  | Sangoma   |
| weakpasswords       | 17.0.1     | Enabled | GPLv3+  | Sangoma   |
+---------------------+------------+---------+---------+-----------+```
```
### Relevant log output

```
[2024-10-11 22:44:59] VERBOSE[537837][C-00000001] res_agi.c: AGI Tx >> agi_network: yes
[2024-10-11 22:44:59] VERBOSE[537837][C-00000001] res_agi.c: AGI Tx >> agi_network_script: enumlookup.agi
[2024-10-11 22:44:59] VERBOSE[537837][C-00000001] res_agi.c: <IAX2/5474-7076>AGI Tx >> agi_request: agi://127.0.0.1/enumlookup.agi
[2024-10-11 22:44:59] VERBOSE[537837][C-00000001] res_agi.c: <IAX2/5474-7076>AGI Tx >> agi_channel: IAX2/5474-7076
[2024-10-11 22:44:59] VERBOSE[537837][C-00000001] res_agi.c: <IAX2/5474-7076>AGI Tx >> agi_language: en_GB
[2024-10-11 22:44:59] VERBOSE[537837][C-00000001] res_agi.c: <IAX2/5474-7076>AGI Tx >> agi_type: IAX2
[2024-10-11 22:44:59] VERBOSE[537837][C-00000001] res_agi.c: <IAX2/5474-7076>AGI Tx >> agi_uniqueid: 1728683099.0
[2024-10-11 22:44:59] VERBOSE[537837][C-00000001] res_agi.c: <IAX2/5474-7076>AGI Tx >> agi_version: 20.9.3~dfsg+~cs6.14.60671435-1~bpo12+1
[2024-10-11 22:44:59] VERBOSE[537837][C-00000001] res_agi.c: <IAX2/5474-7076>AGI Tx >> agi_callerid: 495474
[2024-10-11 22:44:59] VERBOSE[537837][C-00000001] res_agi.c: <IAX2/5474-7076>AGI Tx >> agi_calleridname: José Gonçalves (Ext)
[2024-10-11 22:44:59] VERBOSE[537837][C-00000001] res_agi.c: <IAX2/5474-7076>AGI Tx >> agi_callingpres: 1
[2024-10-11 22:44:59] VERBOSE[537837][C-00000001] res_agi.c: <IAX2/5474-7076>AGI Tx >> agi_callingani2: 0
[2024-10-11 22:44:59] VERBOSE[537837][C-00000001] res_agi.c: <IAX2/5474-7076>AGI Tx >> agi_callington: 0
[2024-10-11 22:44:59] VERBOSE[537837][C-00000001] res_agi.c: <IAX2/5474-7076>AGI Tx >> agi_callingtns: 0
[2024-10-11 22:44:59] VERBOSE[537837][C-00000001] res_agi.c: <IAX2/5474-7076>AGI Tx >> agi_dnid: 21841XXXX
[2024-10-11 22:44:59] VERBOSE[537837][C-00000001] res_agi.c: <IAX2/5474-7076>AGI Tx >> agi_rdnis: unknown
[2024-10-11 22:44:59] VERBOSE[537837][C-00000001] res_agi.c: <IAX2/5474-7076>AGI Tx >> agi_context: macro-dialout-enum
[2024-10-11 22:44:59] VERBOSE[537837][C-00000001] res_agi.c: <IAX2/5474-7076>AGI Tx >> agi_extension: s
[2024-10-11 22:44:59] VERBOSE[537837][C-00000001] res_agi.c: <IAX2/5474-7076>AGI Tx >> agi_priority: 13
[2024-10-11 22:44:59] VERBOSE[537837][C-00000001] res_agi.c: <IAX2/5474-7076>AGI Tx >> agi_enhanced: 0.0
[2024-10-11 22:44:59] VERBOSE[537837][C-00000001] res_agi.c: <IAX2/5474-7076>AGI Tx >> agi_accountcode: XXXXXXXXXX
[2024-10-11 22:44:59] VERBOSE[537837][C-00000001] res_agi.c: <IAX2/5474-7076>AGI Tx >> agi_threadid: 140317710120640
[2024-10-11 22:44:59] VERBOSE[537837][C-00000001] res_agi.c: <IAX2/5474-7076>AGI Tx >> 
[2024-10-11 22:44:59] VERBOSE[537837][C-00000001] res_agi.c: <IAX2/5474-7076>AGI Rx << GET VARIABLE DIAL_NUMBER
[2024-10-11 22:44:59] VERBOSE[537837][C-00000001] res_agi.c: <IAX2/5474-7076>AGI Tx >> 200 result=1 (35121841XXXX)
[2024-10-11 22:44:59] VERBOSE[537837][C-00000001] res_agi.c: <IAX2/5474-7076>AGI Rx << GET VARIABLE ENUMUSEGOOGLEDNS
[2024-10-11 22:44:59] VERBOSE[537837][C-00000001] res_agi.c: <IAX2/5474-7076>AGI Tx >> 200 result=0
[2024-10-11 22:44:59] VERBOSE[537837][C-00000001] res_agi.c: <IAX2/5474-7076>AGI Rx << VERBOSE "Looking up 35121841XXXX on e164.org via dns_get_record" 3
[2024-10-11 22:44:59] VERBOSE[537837][C-00000001] res_agi.c: <IAX2/5474-7076>AGI Tx >> 200 result=1
[2024-10-11 22:44:59] VERBOSE[537837][C-00000001] res_agi.c: <IAX2/5474-7076>AGI Rx << VERBOSE "Looking up 35121841XXXX on e164.arpa via dns_get_record" 3
[2024-10-11 22:44:59] VERBOSE[537837][C-00000001] res_agi.c: <IAX2/5474-7076>AGI Tx >> 200 result=1
[2024-10-11 22:44:59] VERBOSE[537837][C-00000001] res_agi.c: <IAX2/5474-7076>AGI Rx << VERBOSE "Looking up 35121841XXXX on e164.info via dns_get_record" 3
[2024-10-11 22:44:59] VERBOSE[537837][C-00000001] res_agi.c: <IAX2/5474-7076>AGI Tx >> 200 result=1
[2024-10-11 22:44:59] VERBOSE[537837][C-00000001] res_agi.c: <IAX2/5474-7076>AGI Rx << VERBOSE "Setting DIALARR to sip/21841XXXX@voip.utl.pt%" 3
[2024-10-11 22:44:59] VERBOSE[537837][C-00000001] res_agi.c: <IAX2/5474-7076>AGI Tx >> 200 result=1
[2024-10-11 22:44:59] VERBOSE[537837][C-00000001] res_agi.c: <IAX2/5474-7076>AGI Rx << SET VARIABLE DIALARR "sip/21841XXXX@voip.utl.pt%"
[2024-10-11 22:44:59] VERBOSE[537837][C-00000001] res_agi.c: <IAX2/5474-7076>AGI Tx >> 200 result=1
[2024-10-11 22:44:59] VERBOSE[537837][C-00000001] res_agi.c: <IAX2/5474-7076>AGI Tx >> HANGUP
[2024-10-11 22:44:59] WARNING[537837][C-00000001] channel.c: No channel type registered for 'sip'
[2024-10-11 22:44:59] NOTICE[537837][C-00000001] app_dial.c: Unable to create channel of type 'sip' (cause 66 - Channel not implemented)
[2024-10-11 22:44:59] VERBOSE[537837][C-00000001] app_dial.c: Everyone is busy/congested at this time (1:0/0/1)
```

**Comment by ramarajan222** [comment]
Hi @JoseGoncalves ,

Can you make the dialplan change on  your systemV17 system and let us know its working with PJSIP or not?

Change details
```
[macro-dialout-enum]
exten => s,n,Set(TRYDIAL=${REPLACE(TRYDIAL,/,:)}) ;  for replace / with : 
exten => s,n,Dial(PJSIP/${FROMEXTEN}/${TRYDIAL},${TRUNK_RING_TIMER},${DIAL_TRUNK_OPTIONS}) ; modified to match pjsip syntax
```

To make this change on your system, just copy the below dialplan in to extensions_override_freepbx.conf file.

```
[macro-dialout-enum]
include => macro-dialout-enum-custom
exten => s,1,GosubIf($[$["${ARG3}" != ""] & $["${DB(AMPUSER/${AMPUSER}/pinless)}" != "NOPASSWD"]]?sub-pincheck,s,1())
exten => s,n,GotoIf($["x${OUTDISABLE_${DIAL_TRUNK}}" = "xon"]?disabletrunk,1)
exten => s,n,Set(DIAL_TRUNK_OPTIONS=${IF($["${DB_EXISTS(TRUNK/${DIAL_TRUNK}/dialopts)}" = "1"]?${DB_RESULT}:${TRUNK_OPTIONS})})
exten => s,n,Set(OUTBOUND_GROUP=OUT_${ARG1})
exten => s,n,GotoIf($["${OUTMAXCHANS_${ARG1}}foo" = "foo"]?nomax)
exten => s,n,GotoIf($[ ${GROUP_COUNT(OUT_${ARG1})} >= ${OUTMAXCHANS_${ARG1}} ]?nochans)
exten => s,n(nomax),Set(DIAL_NUMBER=${ARG2})
exten => s,n,Set(DIAL_TRUNK=${ARG1})
exten => s,n,GotoIf($["${INTRACOMPANYROUTE}" = "YES"]?skipoutcid)
exten => s,n,Set(DIAL_TRUNK_OPTIONS=${DIAL_OPTIONS})
exten => s,n,Gosub(macro-outbound-callerid,s,1(${DIAL_TRUNK}))
exten => s,n(skipoutcid),GosubIf($["${PREFIX_TRUNK_${DIAL_TRUNK}}" != ""]?sub-flp-${DIAL_TRUNK},s,1())
exten => s,n,AGI(agi://127.0.0.1/enumlookup.agi)
exten => s,n,ExecIf($["${DB(AMPUSER/${AMPUSER}/cidname)}" != ""]?Set(CONNECTEDLINE(num,i)=${DIAL_NUMBER}))
exten => s,n,ExecIf($["${DB(AMPUSER/${AMPUSER}/cidname)}" != ""]?Set(CONNECTEDLINE(name,i)=CID:${CALLERID(number)}))
exten => s,n(dialloop),GotoIf($["foo${DIALARR}"="foo"]?s-${DIALSTATUS},1)
exten => s,n,ExecIf($["${MOHCLASS}"!="default" & "${MOHCLASS}"!="" & "${FORCE_CONFIRM}"="" ]?Set(DIAL_TRUNK_OPTIONS=M(setmusic^${MOHCLASS})${DIAL_TRUNK_OPTIONS}))
exten => s,n,ExecIf($["${FORCE_CONFIRM}"!="" ]?Set(DIAL_TRUNK_OPTIONS=M(confirm)${DIAL_TRUNK_OPTIONS}))
exten => s,n,Set(TRYDIAL=${CUT(DIALARR,%,1)})
exten => s,n,Set(DIALARR=${CUT(DIALARR,%,2-)})
exten => s,n,Set(TRYDIAL=${REPLACE(TRYDIAL,/,:)})
exten => s,n,Dial(PJSIP/${FROMEXTEN}/${TRYDIAL},${TRUNK_RING_TIMER},${DIAL_TRUNK_OPTIONS})
exten => s,n,GotoIf($[ $[ "${DIALSTATUS}" = "CHANUNAVAIL" ] | $[ "${DIALSTATUS}" = "CONGESTION" ] ]?dialloop)
exten => s,n,GotoIf($["${ARG4}" = "on"]?continue,1:s-${DIALSTATUS},1)
exten => s,n(nochans),Noop(max channels used up)
```


Run dialplan reload in asterisk cli and then test the calls with ENUM trunk.

**Comment by JoseGoncalves** [comment]
Hi @ramarajan222 

It does not work, but now it gives me a different error:

```
[2024-10-18 08:13:38] VERBOSE[1434042][C-000002bf] res_agi.c: <IAX2/5474-10153>AGI Rx << VERBOSE "Setting DIALARR to sip/21841XXXX@voip.utl.pt%" 3
[2024-10-18 08:13:38] VERBOSE[1434042][C-000002bf] res_agi.c: agi://127.0.0.1/enumlookup.agi: Setting DIALARR to sip/21841XXXX@voip.utl.pt%
[2024-10-18 08:13:38] VERBOSE[1434042][C-000002bf] res_agi.c: <IAX2/5474-10153>AGI Tx >> 200 result=1
[2024-10-18 08:13:38] VERBOSE[1434042][C-000002bf] res_agi.c: <IAX2/5474-10153>AGI Rx << SET VARIABLE DIALARR "sip/21841XXXX@voip.utl.pt%"
[2024-10-18 08:13:38] VERBOSE[1434042][C-000002bf] res_agi.c: <IAX2/5474-10153>AGI Tx >> 200 result=1
[2024-10-18 08:13:38] VERBOSE[1434042][C-000002bf] res_agi.c: <IAX2/5474-10153>AGI Script agi://127.0.0.1/enumlookup.agi completed, returning 0
[2024-10-18 08:13:38] VERBOSE[1434042][C-000002bf] res_agi.c: <IAX2/5474-10153>AGI Tx >> HANGUP
[2024-10-18 08:13:38] VERBOSE[1434042][C-000002bf] pbx.c: Executing [s@macro-dialout-enum:14] ExecIf("IAX2/5474-10153", "1?Set(CONNECTEDLINE(num,i)=35121841XXXX)") in new stack
[2024-10-18 08:13:38] VERBOSE[1434042][C-000002bf] pbx.c: Executing [s@macro-dialout-enum:15] ExecIf("IAX2/5474-10153", "1?Set(CONNECTEDLINE(name,i)=CID:495474)") in new stack
[2024-10-18 08:13:38] VERBOSE[1434042][C-000002bf] pbx.c: Executing [s@macro-dialout-enum:16] GotoIf("IAX2/5474-10153", "0?s-,1") in new stack
[2024-10-18 08:13:38] VERBOSE[1434042][C-000002bf] pbx.c: Executing [s@macro-dialout-enum:17] ExecIf("IAX2/5474-10153", "0?Set(DIAL_TRUNK_OPTIONS=M(setmusic^default))") in new stack
[2024-10-18 08:13:38] VERBOSE[1434042][C-000002bf] pbx.c: Executing [s@macro-dialout-enum:18] ExecIf("IAX2/5474-10153", "0?Set(DIAL_TRUNK_OPTIONS=M(confirm))") in new stack
[2024-10-18 08:13:38] VERBOSE[1434042][C-000002bf] pbx.c: Executing [s@macro-dialout-enum:19] Set("IAX2/5474-10153", "TRYDIAL=sip/21841XXXX@voip.utl.pt") in new stack
[2024-10-18 08:13:38] VERBOSE[1434042][C-000002bf] pbx.c: Executing [s@macro-dialout-enum:20] Set("IAX2/5474-10153", "DIALARR=") in new stack
[2024-10-18 08:13:38] VERBOSE[1434042][C-000002bf] pbx.c: Executing [s@macro-dialout-enum:21] Set("IAX2/5474-10153", "TRYDIAL=sip:21841XXXX@voip.utl.pt") in new stack
[2024-10-18 08:13:38] VERBOSE[1434042][C-000002bf] pbx.c: Executing [s@macro-dialout-enum:22] Dial("IAX2/5474-10153", "PJSIP/5474/sip:21841XXXX@voip.utl.pt,300,") in new stack
[2024-10-18 08:13:38] ERROR[1169639] chan_pjsip.c: Unable to create PJSIP channel - endpoint '5474' was not found
[2024-10-18 08:13:38] NOTICE[1434042][C-000002bf] app_dial.c: Unable to create channel of type 'PJSIP' (cause 3 - No route to destination)
[2024-10-18 08:13:38] VERBOSE[1434042][C-000002bf] app_dial.c: Everyone is busy/congested at this time (1:0/0/1)
```

**Comment by ramarajan222** [comment]
HI @JoseGoncalves Is it possible for you to test the call from the PJSIP extension?

**Comment by JoseGoncalves** [comment]
Not now... I'm not on my office, and remotely I can only use IAX2 to make calls... later on the day I will try it when I reach my office.

**Comment by JoseGoncalves** [comment]
Hi @ramarajan222. It works from a PJSIP extension. Here is the relevant log:
```
[2024-10-18 13:22:53] VERBOSE[1466917][C-00000326] res_agi.c: agi://127.0.0.1/enumlookup.agi: Setting DIALARR to sip/21841XXXX@voip.utl.pt%
[2024-10-18 13:22:53] VERBOSE[1466917][C-00000326] res_agi.c: <PJSIP/2474-00000648>AGI Tx >> 200 result=1
[2024-10-18 13:22:53] VERBOSE[1466917][C-00000326] res_agi.c: <PJSIP/2474-00000648>AGI Rx << SET VARIABLE DIALARR "sip/21841XXXX@voip.utl.pt%"
[2024-10-18 13:22:53] VERBOSE[1466917][C-00000326] res_agi.c: <PJSIP/2474-00000648>AGI Tx >> 200 result=1
[2024-10-18 13:22:53] VERBOSE[1466917][C-00000326] res_agi.c: <PJSIP/2474-00000648>AGI Script agi://127.0.0.1/enumlookup.agi completed, returning 0
[2024-10-18 13:22:53] VERBOSE[1466917][C-00000326] res_agi.c: <PJSIP/2474-00000648>AGI Tx >> HANGUP
[2024-10-18 13:22:54] VERBOSE[1466917][C-00000326] pbx.c: Executing [s@macro-dialout-enum:14] ExecIf("PJSIP/2474-00000648", "1?Set(CONNECTEDLINE(num,i)=35121841XXXX)") in new stack
[2024-10-18 13:22:54] VERBOSE[1466917][C-00000326] pbx.c: Executing [s@macro-dialout-enum:15] ExecIf("PJSIP/2474-00000648", "1?Set(CONNECTEDLINE(name,i)=CID:492474)") in new stack
[2024-10-18 13:22:54] VERBOSE[1466917][C-00000326] pbx.c: Executing [s@macro-dialout-enum:16] GotoIf("PJSIP/2474-00000648", "0?s-,1") in new stack
[2024-10-18 13:22:54] VERBOSE[1466917][C-00000326] pbx.c: Executing [s@macro-dialout-enum:17] ExecIf("PJSIP/2474-00000648", "0?Set(DIAL_TRUNK_OPTIONS=M(setmusic^default))") in new stack
[2024-10-18 13:22:54] VERBOSE[1466917][C-00000326] pbx.c: Executing [s@macro-dialout-enum:18] ExecIf("PJSIP/2474-00000648", "0?Set(DIAL_TRUNK_OPTIONS=M(confirm))") in new stack
[2024-10-18 13:22:54] VERBOSE[1466917][C-00000326] pbx.c: Executing [s@macro-dialout-enum:19] Set("PJSIP/2474-00000648", "TRYDIAL=sip/21841XXXX@voip.utl.pt") in new stack
[2024-10-18 13:22:54] VERBOSE[1466917][C-00000326] pbx.c: Executing [s@macro-dialout-enum:20] Set("PJSIP/2474-00000648", "DIALARR=") in new stack
[2024-10-18 13:22:54] VERBOSE[1466917][C-00000326] pbx.c: Executing [s@macro-dialout-enum:21] Set("PJSIP/2474-00000648", "TRYDIAL=sip:21841XXXX@voip.utl.pt") in new stack
[2024-10-18 13:22:54] VERBOSE[1466917][C-00000326] pbx.c: Executing [s@macro-dialout-enum:22] Dial("PJSIP/2474-00000648", "PJSIP/2474/sip:21841XXXX@voip.utl.pt,300,") in new stack
[2024-10-18 13:22:54] VERBOSE[1466917][C-00000326] app_dial.c: Called PJSIP/2474/sip:21841XXXX@voip.utl.pt
[2024-10-18 13:22:54] VERBOSE[1044872] netsock2.c: Using SIP RTP Audio TOS bits 184
[2024-10-18 13:22:54] VERBOSE[1044872] netsock2.c: Using SIP RTP Audio CoS mark 5
[2024-10-18 13:22:54] VERBOSE[1466917][C-00000326] app_dial.c: PJSIP/2474-00000649 is ringing
```

So, the only issue remaining is when I make a call from an IAX2 extension.

**Comment by ramarajan222** [comment]
Thank you for the update @JoseGoncalves, Need to modify the dial command to fix it with IAX2 extension.

**Comment by JoseGoncalves** [comment]
Is there a ETA for a fix for this issue (that works with both PJSIP and IAX2 extensions) ?

**Comment by kapilgupta01** [comment]
hi @JoseGoncalves will try to look into this one asap but can not provide the real ETA. thanks

**Comment by JoseGoncalves** [comment]
Are there any plans to address this issue in the near future?
